galasa-ui/src/components/users/UserRoleSection.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+"use client";
+import React, { useEffect, useState } from 'react';
+import { ProfileDetailsProps } from '@/utils/interfaces';
+import { UserData } from '@/generated/galasaapi';
+import styles from "@/styles/UserRole.module.css";
+import { ButtonSet, Button, Dropdown, Loading } from '@carbon/react';
+import ErrorPage from '@/app/error/page';
+import { InlineNotification } from '@carbon/react';
+
+interface DropdownItem {
+  name: string,
+  description: string
+}
+
+export default function UserRoleSection({ userProfilePromise }: ProfileDetailsProps) {
+
+  const [userProfile, setUserProfile] = useState<UserData>({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [role, setRole] = useState({
+    id: "",
+    name: "",
+    description: ""
+  });
+  const [isSaveBtnDisabled, setIsSaveBtnDisabled] = useState(true);
+  const [isResetBtnDisabled, setIsResetBtnDisabled] = useState(true);
+  const [isToastVisible, setIsToastVisible] = useState(false);
+
+  const items = [
+    {
+      id: "1",
+      name: 'tester',
+      description: "Test developer and runner"
+    }, {
+      id: "2",
+      name: 'admin',
+      description: "Administrator access"
+    }, {
+      id: "0",
+      name: 'deactivated',
+      description: "User has no access"
+    }
+  ];
+
+  useEffect(() => {
+    const loadUserProfile = async () => {
+      setIsError(false);
+      setIsLoading(true);
+
+      try {
+
+        const loadedProfile = await userProfilePromise;
+        setUserProfile(loadedProfile);
+        setRole({ id: loadedProfile.synthetic?.role?.metadata?.id!, name: loadedProfile.synthetic?.role?.metadata?.name!, description: loadedProfile.synthetic?.role?.metadata?.description! });
```

Comment:
Maybe we could just set the `role` to be `roleMetadata` and then we can pick out the ID, name, and description from there rather than copying the metadata out?